app/parse.py

```diff
@@ -15,8 +42,128 @@ class Product:
     num_of_reviews: int
 
 
+class AbstractParser(ABC):
+    @abstractmethod
+    def parse_page(self, url: str, file_name: str) -> list[Product]:
+        pass
+
+
+def create_csv_file(file_name: str, list_of_products: list[Product]) -> None:
+    """
+    Creates csv file with list of products and its attributes
+    """
+    with open(file_name, "w", newline="") as file:
+        writer = csv.writer(file)
+        writer.writerow(
+            ["title", "description", "price", "rating", "num_of_reviews"]
+        )
+        for product in list_of_products:
+            writer.writerow(
+                [
+                    product.title,
+                    product.description,
+                    product.price,
+                    product.rating,
+                    product.num_of_reviews,
+                ]
+            )
+
+
+class ElectronicProductParser(AbstractParser):
+    def __init__(self, driver: webdriver.Chrome) -> None:
+        self.driver = driver
+        self.cookies_accepted = False
+
+    def parse_page(self, url: str, file_name: str) -> list[Product]:
+        self.driver.get(url)
+
+        if not self.cookies_accepted:
+            self.check_accept_cookies()
+
+        self.handle_pagination()
+
+        items = self.driver.find_elements(By.CLASS_NAME, PRODUCT_WRAPPER_CLASS)
+        list_of_products = []
+
+        # tqdm for progress tracking visualization
+        with tqdm(
+            total=len(items), desc="Parsing Products", unit_scale=True
+        ) as pbar:
+            for item in items:
+                title = item.find_element(
+                    By.CLASS_NAME, PRODUCT_TITLE_CLASS
+                ).get_attribute("title")
+                description = item.find_element(
+                    By.CLASS_NAME, PRODUCT_DESCRIPTION_CLASS
+                ).text
+                price = float(
+                    item.find_element(By.CLASS_NAME, PRODUCT_PRICE_CLASS).text[
+                        1:
+                    ]
+                )
+                rating = len(
+                    item.find_elements(By.CLASS_NAME, PRODUCT_RATING_CLASS)
+                )
+                num_of_reviews = int(
+                    item.find_element(
+                        By.CLASS_NAME, PRODUCT_REVIEW_COUNT_CLASS
+                    ).text.split()[0]
+                )
+                list_of_products.append(
+                    Product(title, description, price, rating, num_of_reviews)
+                )
```

Comment:
if it's a class, then it's better to have a separate method for each field that should be scrapped